### PR TITLE
Improvement/pre ga hotfix support

### PR DESCRIPTION
**Pull request by charlesprost**
This pull request introduces several improvements and fixes to the versioning and cascade logic for hotfix and development branches, especially in pre-GA (General Availability) scenarios. The changes ensure that hotfix branches without a GA tag are handled correctly, development branches target the right versions when hotfixes exist, and the test suite covers these edge cases. Additionally, minor improvements to git configuration and quoting logic are included.

**Hotfix and Development Branch Cascade Logic Improvements:**

* Hotfix branches now default to version `x.y.z.0` before a GA tag exists, and this is reflected in the new `hfrev = 0` default and initialization logic in `HotfixBranch`, ensuring pre-GA hotfixes are versioned correctly.
* Development branches (e.g., `development/10`) now correctly advance their target version (e.g., to `10.1.0`) when a pre-GA hotfix branch (e.g., `hotfix/10.0.0`) exists, even if no GA tag is present. This is achieved by tracking "phantom" hotfixes for version calculation without including them in the actual cascade. [[1]](diffhunk://#diff-2f6e1179ed4541ba7239ae873ce6ad8cf45355d5dd9e02a2de9cad20b3b436faR868) [[2]](diffhunk://#diff-2f6e1179ed4541ba7239ae873ce6ad8cf45355d5dd9e02a2de9cad20b3b436faR938-R946) [[3]](diffhunk://#diff-2f6e1179ed4541ba7239ae873ce6ad8cf45355d5dd9e02a2de9cad20b3b436faL1074-R1105) [[4]](diffhunk://#diff-2f6e1179ed4541ba7239ae873ce6ad8cf45355d5dd9e02a2de9cad20b3b436faL1240-R1265)

**Test Suite Enhancements:**

* Added comprehensive tests for pre-GA hotfix and development branch scenarios, including: ensuring correct fix versioning for PRs targeting hotfix branches before GA, verifying development branches target the correct versions when pre-GA hotfixes exist, and handling mixed cascades with 2-digit and 3-digit development branches. [[1]](diffhunk://#diff-5c0d206b167ef78f6bc457ba29b9c02058cbdc8b7fbe7239f243b98f1e06590fR444-R447) [[2]](diffhunk://#diff-5c0d206b167ef78f6bc457ba29b9c02058cbdc8b7fbe7239f243b98f1e06590fR462-R516) [[3]](diffhunk://#diff-5c0d206b167ef78f6bc457ba29b9c02058cbdc8b7fbe7239f243b98f1e06590fR5961-R5982)

**Queue and Build Logic Adjustments:**

* Updated queue integration logic to allow hotfix branches with `hfrev >= 0` (pre-GA or post-GA) to use their version for the queue branch, ensuring correct queue branch naming and Jira fix version assignment.
* Fixed build status retrieval in the cascade logic to use the latest commit for accurate status checks.

**Other Improvements:**

* Switched from the deprecated `pipes.quote` to `shlex.quote` for shell quoting, improving compatibility and security.
* Configured git to always use fast-forward merges by default in `clone_git_repo`, ensuring consistent merge behavior.

### CODE REVIEW

**Comment by charlesprost** [comment]
@claude[agent] review

File: bert_e/tests/test_bert_e.py [hunk 1]
```diff
@@ -455,6 +459,87 @@ def test_branch_cascade_target_hotfix(self):
...
+            4: {'name': 'development/10.1', 'ignore': False},
+            5: {'name': 'development/10', 'ignore': False},
+        })
+        # Pre-GA: hotfix/10.0.0 exists but no 10.0.0.X tag yet.
+        # 3-digit branches (9.5.3, 10.0.1) target their own version because
+        # there are no ancestor 2-digit branches (dev/9.5, dev/10.0) in the
+        # cascade to drive _next_micro.
+        # dev/10.1 → 10.1.0, dev/10 → 10.2.0 (latest_minor=1 from dev/10.1).
+        tags = ['9.5.0', '9.5.1']
+        fixver = ['9.5.3', '10.0.1', '10.1.0', '10.2.0']
```
Comment (oliviergaraud) [inline bert_e/tests/test_bert_e.py:509]: having fixversion hardcoded like this seems brittle to me but 🤷

Comment (charlesprost) [reply bert_e/tests/test_bert_e.py:509]: it's necessary to have them because that's how we can check if it's going to be processed the right way
basically, a test is: "I have those branches, I have those tags, what are the fix versions supposed to be valid for each branch?"

**Review by oliviergaraud — APPROVED** [review]

**Review by matthiasL-scality — APPROVED** [review]

**Review by rdebay-scality — APPROVED** [review]
